eng/scripts/docs/Docs-ToC.ps1

```diff
@@ -0,0 +1,88 @@
+        # Package has an '@' symbol deliminting the end of the package name
+        $packageName = $packageName.Substring(0, $packageName.LastIndexOf('@'))
+      }
+      $onboardedPackages[$packageName] = $null
+    }
+  }
+
+  return $onboardedPackages
+}
+
+function Get-javascript-DocsMsTocData($packageMetadata, $docRepoLocation) {
+  # Fallback to get package-level readme name if metadata file info does not exist
+  $packageLevelReadmeName = $packageMetadata.Package.Replace('@azure/', '').Replace('@azure-tools/', '').Replace('azure-', '');
+
+  # Fallback to get package-level readme name if metadata file info does not exist
+  if ($packageMetadata.Package.StartsWith('@azure-rest/')) {
+    $packageLevelReadmeName = "$($packageMetadata.Package.Replace('@azure-rest/', ''))-rest"
+  }
+
+  # If there is a metadata json for the package use the DocsMsReadmeName from
+  # the metadata function
+  if ($packageMetadata.PSObject.Members.Name -contains "FileMetadata") {
+    $readmeMetadata = &$GetDocsMsMetadataForPackageFn -PackageInfo $packageMetadata.FileMetadata
+    $packageLevelReadmeName = $readmeMetadata.DocsMsReadMeName
+  }
+
+
+  $packageTocHeader = $packageMetadata.Package
+  if ($clientPackage.DisplayName) {
+    $packageTocHeader = $clientPackage.DisplayName
+  }
+  $output = [PSCustomObject]@{
+    PackageLevelReadmeHref = "~/docs-ref-services/{moniker}/$packageLevelReadmeName-readme.md"
+    PackageTocHeader       = $packageTocHeader
+    TocChildren            = @($clientPackage.Package)
+  }
+
+  return $output
+}
+
+function Get-javascript-DocsMsTocChildrenForManagementPackages($packageMetadata, $docRepoLocation) {
+  return @($packageMetadata.Package)
+}
+
+function Get-javascript-UpdatedDocsMsToc($toc) {
+  $items = $toc[0].items
+  for ($i = 0; $i -lt $items.Count; $i++) {
+
+    # Add "Plugin" docs to Identity. Packages associated with these entries do
+    # not build successfully in the docs CI system becaues they export nothing
+    # that the docs CI system can document. This ensures that the readme pages
+    # are documented properly even if their packages are not onboarded.
+    if ($items[$i].name -eq 'Identity') {
+      $items[$i].items += [PSCustomObject]@{
+        name  = "Plugins";
+        items = @(
+          [PSCustomObject]@{
+            name            = "Token Cache Persistence";
+            href            = "~/docs-ref-services/{moniker}/identity-cache-persistence-readme.md";
+            landingPageType = "Service";
+          },
+          [PSCustomObject]@{
+            name            = "VSCode Authentication";
+            href            = "~/docs-ref-services/{moniker}/identity-vscode-readme.md";
+            landingPageType = "Service";
+          }
+        )
+      }
+    }
+  }
+
+  # PowerShell outputs a single object if the output is an array with only one
+  # object. The preceeding comma ensures that the output remains an array for
+  # appropriate export formatting.
+  return , $toc
```

Comment:
Similar to how you handle it above, making to code clearer so no need for a comment.

```suggestion
  return @($toc)
```

Reply:
The proposed change does not produce the correct YAML output but the existing code does. I've added to the comment here.